### PR TITLE
Linux thread types

### PR DESCRIPTION
**Pull request by arnetheduck**
needed for nlvm

### CODE REVIEW

File: lib/system/threads.nim [hunk 1]
```diff
@@ -178,7 +183,12 @@ else:
                                size: int): cint {.
       importc: "pthread_attr_setstack", header: pthreadh.}
 
-  type CpuSet {.importc: "cpu_set_t", header: schedh.} = object
+  when defined(linux) and defined(amd64):
+    type CpuSet {.importc: "cpu_set_t", header: schedh.} = object
+      data: array[1024/(8*sizeof(clong))]
```
Comment (Araq) [inline lib/system/threads.nim]: shouldn't that be 'div' here? and also have an array type too?

Comment (arnetheduck) [inline lib/system/threads.nim]: oops. fixed.

**Comment by Araq** [comment]
Seems overly complex for the purpose. Why not give these objects a ``payload: array[someSize, int]`` instead? This way you can even do it platform independently by choosing a size that's enough for every Posix'ish OS.

**Comment by arnetheduck** [comment]
That would be preferable, yeah, then the same thing would work across all linuxes at least, and it would be easy to add 32-bit and non-x86 because of the way their laid out (I wouldn't have any hopes for other platforms though..).. 

But there are two issues:
1) those fields contain useful debugging information (useful, but not critical..), specially when a deadlock happens
2) nim insists on generating rtti for the types, so the nim names / fields have to match what's in the header file - any way around that?

**Comment by Araq** [comment]
Sure, introduce a .head pragma that is ignored for ``-d:llvm`` (or whatever your define is) and is a synonym for .header otherwise.

**Comment by arnetheduck** [comment]
not sure I follow - the problem is with nim, not nlvm (nlvm ignores header pragmas).. nim compiles rtti for the types which makes use of the offsetof macro (nlvm calculates offsets based on the type) and thus the names in nim and in the header file must match. 

Could simply remove the header pragma for nim, but there's no way to pass alignment info from what I can tell, so that would potentially break some of those types (as the patch is done, the alignment info is read from the c header.. and is broken for nlvm, but I can work around that for now..)

**Comment by arnetheduck** [comment]
replaced by #5018 that takes the array approach

**Pull request closed without merging**